profiler/profiler.go

```diff
@@ -151,25 +161,29 @@ func (p *profiler) enqueueUpload(bat batch) {
 
 // send takes profiles from the output queue and uploads them.
 func (p *profiler) send() {
-	defer close(p.exit)
 	for bat := range p.out {
+		if p.uploadFunc == nil {
```

Comment:
Why do we need this check? It should never be empty. If it is, tests should catch it.